var/spack/repos/builtin/packages/libxsmm/package.py

```diff
@@ -61,6 +61,14 @@ class Libxsmm(Package):
     variant('header-only', default=False,
             description='Produce header-only installation')
 
+    @property
+    def libs(self):
+        result = find_libraries(['libxsmm', 'libxsmmf'], root=self.prefix,
+                                recurse=True)
+        result += find_libraries(['libxsmm', 'libxsmmf'], root=self.prefix,
+                                shared=False, recurse=True)
```

Comment:
Why do u need both shared and static?

Reply:
It's because the library can be built with shared and/or static versions -- what's the best way to handle this?

Reply:
> what's the best way to handle this?

Different packages do it differently, some have `shared` variant which is used in build system and in `libs()`, some just return shared always.

But it certainly strange to return both shared and static together here. You can at least try searching for static if shared were not found.

Reply:
Done.

Reply:
For now I'd add a `shared` variant that defaults true, and return either shared libs or static depending on what it's set to.

@alalazo: this is probably another place where the interface could come in handy.  The libs you want probably depend on whether the caller is built shared. Might be worth thinking about how to integrate that after SC.

Reply:
@tgamblin
> Might be worth thinking about how to integrate that after SC.

Definitely. Probably we also need to decide on a solution for the problem raised in #5269 that works for every use case we have. Lately I am not pushing new features because I am waiting for the release to be tagged, and don't want to add noise to that.

@baip I think adding a variant:
```python
 variant('shared', default=True, description="Build shared libraries")
```
as @tgamblin suggested, and then having:
```python
@property
def libs(self):
    shared = True if '+shared' in self.spec else False
    return find_libraries(
        ['libxsmm', 'libxsmmf'], root=self.prefix, shared=shared, recurse=True
    )
```
is more along the lines of what we do in other packages, at least for now